src/logic/data-access/cookies-provider.ts

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+export class CookiesProvider {
+  public static getCookie(key: string): string | undefined {
+    try {
+      const filteredCookie = document.cookie
+        .split(";")
+        .filter((cookie) => cookie.split("=")[0].trim() === key);
+      const cookieValue = filteredCookie[0].split("=")[1];
+      return cookieValue;
+    } catch (err) {
+      return;
```

Comment:
Robisz tutaj `catch` na coś w w stylu 
`document.cookie.split -cannot split value of undefined`?
Jeśli tak, to lepiej użyc pytajnika, który sprawdza na to czy obiekt przed nim jest nullem i uruchamia kod tylko jeśli obiekt nullem nie jest.
Wtedy metoda wyglądała by mniej więcej tak:
```ts
 const filteredCookie = document.cookie?.split(";")
         .filter((cookie) => cookie.split("=")[0]?.trim() === key);
 const cookieValue = filteredCookie[0]?.split("=")?.[1];
 return cookieValue;
```
I `cookie value` było by albo undefindem albo wartością w zależności od tego jakie wartości mają inne zmienne.

Według mnie, pytajnik ma taką zaletę, że wtedy lepiej widać logikę tego co się dzieje w funkcji.